Harvesting, planting and seed saving modal forms

Br3nda opened the pull request:
When something has not been harvested before:
![image](https://user-images.githubusercontent.com/12266/75293978-30771e00-588c-11ea-9bcd-95c7e21d22b3.png)


And when it has:
![image](https://user-images.githubusercontent.com/12266/75294011-408efd80-588c-11ea-8eb5-f62b32d9aef5.png)

simicic commented on app/helpers/icons_helper.rb:
```diff
@@ -91,6 +91,10 @@ def like_icon
     icon('fas', 'heart')
   end
 
+  def plantpart_icon(name)
```
```suggestion
  def plant_part_icon(name)
```
Simply for the consistency, not a biggie...

simicic commented on app/views/crops/_thumbnail.html.haml:
```diff
@@ -1,9 +1,10 @@
   .card.crop-thumbnail
-    = link_to image_tag(crop.thumbnail_url ? crop.thumbnail_url : placeholder_image,
+    = link_to crop_path(slug: crop.slug) do
+      = image_tag(crop.thumbnail_url ? crop.thumbnail_url : placeholder_image,
```
```suggestion
      = image_tag(crop.thumbnail_url.presence || placeholder_image,
```

simicic commented on app/views/harvests/_modal.html.haml:
```diff
@@ -0,0 +1,30 @@
+              .card
+                = link_to harvests_path(harvest: {crop_id: @crop.id, plant_part_id: plant_part.id}), method: :post, class: 'dropdown-item' do
+                  .card-title.text-center
+                    %h3= plantpart_icon(plant_part.name)
```
```suggestion
                    %h3= plant_part_icon(plant_part.name)
```

simicic commented on app/views/harvests/_modal.html.haml:
```diff
@@ -0,0 +1,30 @@
+        %nav
+          - PlantPart.all.order(:name).each do |plant_part|
+            = link_to harvests_path(harvest: {crop_id: @crop.id, plant_part_id: plant_part.id}), method: :post, class: 'nav-link' do
+              = plantpart_icon(plant_part.name)
```
```suggestion
              = plant_part_icon(plant_part.name)
```

simicic commented on app/views/harvests/_modal.html.haml:
```diff
@@ -0,0 +1,30 @@
+          %hr/
+        %h6 All Plant parts
+        %nav
+          - PlantPart.all.order(:name).each do |plant_part|
```
Would be maybe a good idea to move this to the controller? So it is clear from the controller method what data is necessary to build the view - it might improve understanding.

Br3nda replied:
If it's queried in the view, then we can cache it as part of the view (and have one less database query on the page)

Br3nda replied:
i've added caching

Br3nda commented:
Changed layout, and added to planting pages too
![image](https://user-images.githubusercontent.com/12266/75586672-d23f7a80-5ad9-11ea-86e9-062d4c5f8613.png)

Br3nda commented:
![Screenshot_20200229_100055_com android chrome](https://user-images.githubusercontent.com/12266/75587044-7aedda00-5ada-11ea-8ea9-8f2f3f972ace.jpg)
![Screenshot_20200229_100030_com android chrome](https://user-images.githubusercontent.com/12266/75587051-7de8ca80-5ada-11ea-945a-f63ec209f950.jpg)

Br3nda commented:
modal planting

![image](https://user-images.githubusercontent.com/12266/75603261-a6e87a00-5b31-11ea-9cc6-4e1d5bc2c8e7.png)

Br3nda commented:
seed modal 
![image](https://user-images.githubusercontent.com/12266/75603271-b9fb4a00-5b31-11ea-8d50-d4b02f077278.png)

Br3nda commented on app/assets/images/icons/plant_parts/root.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" width="100px" height="100px"><path fill="#388e3c" d="M42.6,18.9c0,0-0.1-0.1-0.1-0.1c-0.5-0.6,0.2-0.1,0.4-0.4c0.2-0.3-0.5-0.9-0.7-1.1 c-0.4-0.3-0.8-0.5-1.3-0.6c0.2,0,0.6,0,0.6-0.2c0-0.2-0.6-0.5-0.8-0.6c-0.4-0.2-0.8-0.4-1.2-0.6c-0.4-0.1-0.7-0.1-1.1-0.1 c0.1-0.1,0.3-0.1,0.4-0.3c-0.4-0.6-2.2-0.3-2.7,0.1c0.2-0.2,0.3-0.5-0.1-0.5c-0.3,0-0.6,0.1 ...
```
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" width="100px" height="100px"><desc>root</desc><path fill="#388e3c" d="M42.6,18.9c0,0-0.1-0.1-0.1-0.1c-0.5-0.6,0.2-0.1,0.4-0.4c0.2-0.3-0.5-0.9-0.7-1.1 c-0.4-0.3-0.8-0.5-1.3-0.6c0.2,0,0.6,0,0.6-0.2c0-0.2-0.6-0.5-0.8-0.6c-0.4-0.2-0.8-0.4-1.2-0.6c-0.4-0.1-0.7-0.1-1.1-0.1 c0.1-0.1,0.3-0.1,0.4-0.3c-0.4-0.6-2.2-0.3-2.7,0.1c0.2-0.2,0.3-0.5-0.1-0.5c-0.3,0-0.6,0.1-0.9,0.2c-4.2,1.3-7,4.2-7.9,5.2 c-0.9,1-1.3,1.9-1.3,1.9c1.2-1.2,2.7-2.3,4.3-3.1c1-0.5,2.8-1.2,3.9-0.5c0.3,0.2,0.5,0.4,0.8,0.4c0-0.1,0.1-0.2,0.1-0.3 c0.3,0.3,0,0.8,0.3,1.2c0.4,0.5,0.7,0.3,1.1,0c-0.1,0.4,0,1,0.3,1.4c0.3,0.4,0.8,0.4,0.9-0.1c0.1,0.5,0.7,2.3,1.5,1.5 c0.5-0.5,0.8-1.6,0.5-2.2c0.3,0.4,0.6,1.1,1,1.4c0.4,0.3,0.8,0.3,1.3,0.1c0.6-0.4,0.6-1,0.6-1.6c0-0.4,0.1-0.1,0.3-0.2 C43.2,19.6,42.8,19.1,42.6,18.9z"/><path fill="#388e3c" d="M29.1,5.4C29.2,5.4,29.2,5.5,29.1,5.4c0.7,0.6,0.2-0.1,0.5-0.3c0.3-0.2,0.9,0.5,1.1,0.7 c0.3,0.4,0.5,0.8,0.6,1.3c0-0.2,0-0.6,0.2-0.6c0.2,0,0.5,0.6,0.6,0.8c0.2,0.4,0.4,0.8,0.6,1.2c0.1,0.4,0.1,0.7,0.1,1.1 c0.1-0.1,0.1-0.3,0.3-0.4c0.6,0.4,0.3,2.2-0.1,2.7c0.2-0.2,0.5-0.3,0.5,0.1c0,0.3-0.1,0.6-0.2,0.9c-1.3,4.2-4.2,7-5.2,7.9 c-1,0.9-1.9,1.3-1.9,1.3c1.2-1.2,2.3-2.7,3.1-4.3c0.5-1,1.2-2.8,0.5-3.9c-0.2-0.3-0.4-0.5-0.4-0.8c0.1,0,0.2-0.1,0.3-0.1 c-0.3-0.3-0.8,0-1.2-0.3c-0.5-0.4-0.3-0.7,0-1.1c-0.4,0.1-1,0-1.4-0.3c-0.4-0.3-0.4-0.8,0.1-0.9c-0.5-0.1-2.3-0.7-1.5-1.5 c0.5-0.5,1.6-0.8,2.2-0.5c-0.3-0.3-1.1-0.6-1.4-1c-0.3-0.4-0.3-0.8-0.1-1.3c0.4-0.6,1-0.6,1.6-0.6c0.4,0,0.1-0.1,0.2-0.3 C28.4,4.8,28.9,5.2,29.1,5.4z"/><path fill="#4caf50" d="M41.6,6.4c-0.8-0.8-1.6-0.3-2.3,0.4c0.2-0.3,0-0.6-0.4-0.7c-0.3-0.1-0.6,0.1-0.8,0.3 c-0.4,0.4-0.4,1.1-0.4,1.6c-0.3-0.7-1.1-0.4-1.4,0.2c-0.2,0.5-0.2,1.1-0.2,1.6c-0.3-0.3-0.7-0.3-1.1,0.1c-1,1.1-1.1,4.1-6.1,8.9 c-0.1,0.1,0.2,0.3,0.2,0.3c6.1-5.9,7.8-5,9.1-6c0.3-0.3,0-0.5-0.3-0.6c0.4,0,0.8-0.1,1.1-0.1c0.5-0.1,1-0.2,1.4-0.5 c0.2-0.2,0.4-0.6,0.3-0.9c-0.2-0.3-0.5-0.3-0.8-0.3c0.6-0.1,1.3-0.2,1.8-0.6C42,9.6,42.1,9.3,42.1,9c0-0.2-0.2-0.7-0.5-0.6 C42.1,7.9,42.1,6.9,41.6,6.4z"/><path fill="#ff9800" d="M31,27.1c2-2,1.9-5.5-1.4-8.7C26.4,15.1,23,15,21,17C16.7,21.7,5.9,39.8,7.1,40.9 C8.1,41.9,24.7,32.4,31,27.1z"/><path fill="#ef6c00" d="M19.9 28.4c.2.2.4.3.7.3.3 0 .5-.1.7-.3.4-.4.4-1 0-1.4l-4.5-4.5c-.4.6-.8 1.1-1.1 1.7L19.9 28.4zM19.5 18.8l1.9 1.9c.2.2.5.3.7.3.3 0 .5-.1.7-.3.4-.4.4-1 0-1.4l-2.1-2.1C20.4 17.7 20 18.2 19.5 18.8zM9.4 36.9c.2.2.5.3.7.3.3 0 .5-.1.7-.3.4-.4.4-1 0-1.4l-1.2-1.2c-.4.7-.7 1.3-1 1.9L9.4 36.9zM26.9 25.3c-.4-.4-1-.4-1.4 0s-.4 1 0 1.4l2.6 2.6c.6-.4 1.1-.8 1.6-1.2L26.9 25.3zM17 35c-.3 0-.5-.1-.7-.3l-3.1-3.1c-.4-.4-.4-1 0-1.4s1-.4 1.4 0l3.1 3.1c.4.4.4 1 0 1.4C17.5 34.9 17.3 35 17 35z"/></svg>
```

Br3nda commented on app/assets/images/icons/plant_parts/seed.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" width="100px" height="100px"><path fill="#E0AC6E" d="M36 19.999L16 6 4 22.999 25 36.999 31 36.999 38.072 25.394z"/><path fill="#5D4037" d="M36.484 35.725c1.26 1.201 2.741 1.631 3.299.953.562-.677-.008-2.203-1.269-3.404s-2.738-1.631-3.299-.952C34.654 32.996 35.225 34.522 36.484 35.725zM30.359 38.434c1.616.77 3.221.751 3.571-.042.355-.794- ...
```
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" width="100px" height="100px"><desc>seed</desc><path fill="#E0AC6E" d="M36 19.999L16 6 4 22.999 25 36.999 31 36.999 38.072 25.394z"/><path fill="#5D4037" d="M36.484 35.725c1.26 1.201 2.741 1.631 3.299.953.562-.677-.008-2.203-1.269-3.404s-2.738-1.631-3.299-.952C34.654 32.996 35.225 34.522 36.484 35.725zM30.359 38.434c1.616.77 3.221.751 3.571-.042.355-.794-.673-2.063-2.29-2.828-1.616-.768-3.218-.752-3.57.043C27.715 36.398 28.743 37.665 30.359 38.434zM29.484 31.725c1.26 1.201 2.741 1.631 3.299.953.562-.677-.008-2.203-1.269-3.404s-2.738-1.631-3.299-.952C27.654 28.996 28.225 30.522 29.484 31.725zM40 39.999A3 1.5 0 1 0 40 42.999 3 1.5 0 1 0 40 39.999z"/><path fill="#FFCC80" d="M16 6L18.215 11.653 10 22.999 4 22.999 25 36.999 36 19.999z"/><path fill="#558B2F" d="M28,23c0,3.315-2.684,5.999-6,5.999S16,26.315,16,23c0-3.317,2.684-6.001,6-6.001S28,19.683,28,23"/><g><path fill="#FFF" d="M20.536 19.943c0-.829.655-1.498 1.464-1.498s1.464.669 1.464 1.498c0 .832-1.464 3.002-1.464 3.002S20.536 20.775 20.536 19.943M23.464 25.945c0 .831-.655 1.5-1.464 1.5s-1.464-.669-1.464-1.5 1.464-3 1.464-3S23.464 25.114 23.464 25.945"/><path fill="#FFF" d="M18.732 22.743c-.701-.415-.939-1.329-.536-2.049.403-.719 1.3-.964 2.001-.547C20.896 20.562 22 22.945 22 22.945S19.433 23.156 18.732 22.743M25.268 23.146c.701.415.939 1.331.536 2.049-.403.72-1.3.964-2.001.549C23.103 25.33 22 22.945 22 22.945S24.567 22.731 25.268 23.146"/><path fill="#FFF" d="M25.268 22.743c.701-.415.939-1.329.536-2.049-.403-.719-1.3-.964-2.001-.547C23.103 20.562 22 22.945 22 22.945S24.567 23.156 25.268 22.743M18.732 23.146c-.701.415-.939 1.331-.536 2.049.403.72 1.3.964 2.001.549C20.896 25.33 22 22.945 22 22.945S19.433 22.731 18.732 23.146"/></g><path fill="#FFC107" d="M23.5,23c0,0.828-0.671,1.499-1.5,1.499S20.5,23.828,20.5,23c0-0.83,0.671-1.501,1.5-1.501S23.5,22.17,23.5,23"/></svg>
```

Br3nda commented on app/assets/images/icons/plant_parts/stem.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" width="100px" height="100px"><path fill="#558B2F" d="M22,16v28c0,1.1-0.9,2-2,2s-2-0.9-2-2V16H22z M26,16v28c0,1.1,0.9,2,2,2s2-0.9,2-2V16H26z"/><path fill="#1B5E20" d="M20,5c0,0,3,3.8,3,8.5S21.7,22,20,22s-3-3.8-3-8.5S20,5,20,5z M25,13.5c0,4.7,1.3,8.5,3,8.5s3-3.8,3-8.5S28,5,28,5S25,8.8,25,13.5z"/><path fill="#8BC34A" d="M26,16v28c0,1.1-0.9, ...
```
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" width="100px" height="100px"><desc>stem</desc><path fill="#558B2F" d="M22,16v28c0,1.1-0.9,2-2,2s-2-0.9-2-2V16H22z M26,16v28c0,1.1,0.9,2,2,2s2-0.9,2-2V16H26z"/><path fill="#1B5E20" d="M20,5c0,0,3,3.8,3,8.5S21.7,22,20,22s-3-3.8-3-8.5S20,5,20,5z M25,13.5c0,4.7,1.3,8.5,3,8.5s3-3.8,3-8.5S28,5,28,5S25,8.8,25,13.5z"/><path fill="#8BC34A" d="M26,16v28c0,1.1-0.9,2-2,2s-2-0.9-2-2V16H26z M30,16v28c0,1.1,0.9,2,2,2s2-0.9,2-2V16H30z M14,16v28c0,1.1,0.9,2,2,2c1.1,0,2-0.9,2-2V16H14z"/><path fill="#689F38" d="M32,2c0,0,3,3.8,3,8.5S33.7,19,32,19c-1.7,0-3-3.8-3-8.5S32,2,32,2z M13,10.5c0,4.7,1.3,8.5,3,8.5s3-3.8,3-8.5S16,2,16,2S13,5.8,13,10.5z M21,10.5c0,4.7,1.3,8.5,3,8.5s3-3.8,3-8.5S24,2,24,2S21,5.8,21,10.5z"/><path fill="#2E7D32" d="M17,7c0,0.6-0.4,1-1,1c-0.6,0-1-0.4-1-1s1-3,1-3S17,6.4,17,7z M14,7c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S14,7,14,7z M18,7c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S18,7,18,7z M16,10c0,0-1,2.4-1,3s0.4,1,1,1c0.6,0,1-0.4,1-1S16,10,16,10z M13.4,14.7c0.2,0.9,0.4,1.7,0.7,2.3c0.5,0,1-0.5,1-1c0-0.6-1-3-1-3S13.7,13.9,13.4,14.7z M18,13c0,0-1,2.4-1,3c0,0.5,0.4,0.9,0.9,1c0.3-0.7,0.5-1.5,0.7-2.4C18.4,13.8,18,13,18,13z M16,16c0,0-1,2.4-1,3s0.4,1,1,1c0.6,0,1-0.4,1-1S16,16,16,16z M24,4c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S24,4,24,4z M22,7c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S22,7,22,7z M26,7c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S26,7,26,7z M24,10c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S24,10,24,10z M21.4,14.7c0.2,0.9,0.4,1.7,0.7,2.3c0.5,0,1-0.5,1-1c0-0.6-1-3-1-3S21.7,13.9,21.4,14.7z M26,13c0,0-1,2.4-1,3c0,0.5,0.4,0.9,0.9,1c0.3-0.7,0.5-1.5,0.7-2.4C26.4,13.8,26,13,26,13z M24,16c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S24,16,24,16z M32,4c0,0-1,2.4-1,3s0.4,1,1,1c0.6,0,1-0.4,1-1S32,4,32,4z M30,7c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S30,7,30,7z M34,7c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S34,7,34,7z M32,10c0,0-1,2.4-1,3s0.4,1,1,1c0.6,0,1-0.4,1-1S32,10,32,10z M29.4,14.7c0.2,0.9,0.4,1.7,0.7,2.3c0.5,0,1-0.5,1-1c0-0.6-1-3-1-3S29.7,13.9,29.4,14.7z M34,13c0,0-1,2.4-1,3c0,0.5,0.4,0.9,0.9,1c0.3-0.7,0.5-1.5,0.7-2.4C34.4,13.8,34,13,34,13z M32,16c0,0-1,2.4-1,3s0.4,1,1,1c0.6,0,1-0.4,1-1S32,16,32,16z"/><path fill="#558B2F" d="M17,26c0,0.6-0.4,1-1,1c-0.6,0-1-0.4-1-1s1-3,1-3S17,25.4,17,26z M24,23c0,0-1,2.4-1,3s0.4,1,1,1s1-0.4,1-1S24,23,24,23z M32,23c0,0-1,2.4-1,3s0.4,1,1,1c0.6,0,1-0.4,1-1S32,23,32,23z"/></svg>
```

Br3nda replied:
adding descriptions
